Dockerfile

```diff
@@ -1,33 +1,34 @@
-FROM alpine:3.7
+FROM alpine:latest
 
 LABEL maintainer="lucebac <docker@lucebac.net>"
 
 RUN apk add -U --no-cache curl unzip nodejs nodejs-npm sqlite openssl git python file \
     && adduser -D ctfpad \
-    && mkdir /ctfpad && chown ctfpad:ctfpad /ctfpad
+    && mkdir /ctfpad && chown ctfpad:ctfpad /ctfpad \
+    && chown ctfpad:ctfpad -R /ctfpad
 
 WORKDIR /ctfpad
 
 # setup ctfpad
 RUN cd /ctfpad \
     && git clone https://github.com/ENOFLAG/CTFPad ctfpad \
     && cd ctfpad \ 
-    && npm install
+    && npm install \
```

Comment:
Why aren't you using `USER ctfpad` to temporarily change to the correct user? This would probably save a few `chown`s and increse the overall performance.

Reply:
Does npm install work with non-root users? Never used npm before, you can give it a try